ChrootCfg.sh

```diff
@@ -15,23 +15,138 @@
 #   location.
 #
 #####################################
+PROGNAME=$(basename "$0")
 CHROOT="${AMIGENCHROOT:-/mnt/ec2-root}"
+DEBUG="${DEBUG:-UNDEF}"
 EXECIT=${LOCALSCRIPT:-UNDEF}
 NTPCONF="${CHROOT}/etc/ntp.conf"
 CLOUDCF="${CHROOT}/etc/cloud/cloud.cfg"
-TIMEZON="${AMIGENTIMEZONE:-UTC}"
+TIMEZONE="${AMIGENTIMEZONE:-UTC}"
+
+# Make interactive-execution more-verbose unless explicitly told not to
+if [[ $( tty -s ) -eq 0 ]] && [[ ${DEBUG} == "UNDEF" ]]
+then
+   DEBUG="true"
+fi
+
+
+# Error handler function
+function err_exit {
+   local ERRSTR
+   local ISNUM
+   local SCRIPTEXIT
+
+   ERRSTR="${1}"
+   ISNUM='^[0-9]+$'
+   SCRIPTEXIT="${2:-1}"
+
+   if [[ ${DEBUG} == true ]]
+   then
+      # Our output channels
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+# Print out a basic usage message
+function UsageMsg {
+   local SCRIPTEXIT
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-f' 'Filesystem-type of chroot-devs (e.g., "xfs")'
+      printf '\t%-4s%s\n' '-h' 'Print this message'
+      printf '\t%-4s%s\n' '-m' 'Where chroot-dev is mounted (default: "/mnt/ec2-root")'
+      printf '\t%-4s%s\n' '-z' 'Initial timezone of build-target (default: "UTC")'
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--fstype' 'See "-f" short-option'
+      printf '\t%-20s%s\n' '--help' 'See "-h" short-option'
+      printf '\t%-20s%s\n' '--mountpoint' 'See "-m" short-option'
+      printf '\t%-20s%s\n' '--no-tmpfs' 'Disable /tmp as tmpfs behavior'
+      printf '\t%-20s%s\n' '--timezone' 'See "-z" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+######################
+## Main program-flow
+######################
+OPTIONBUFR=$( getopt \
+   -o Ff:hm:z: \
```

Comment:
I don't see these being handled: `-f`, `-F`, `--fstype`, `--no-fips`. What's the plan there?

Reply:
I forgot to remove them.

Reply:
updated